distributed/scheduler.py

```diff
@@ -2126,6 +2126,18 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             n_workers: Py_ssize_t = len(worker_pool_dv)
             if n_workers < 20:  # smart but linear in small case
                 ws = min(worker_pool.values(), key=operator.attrgetter("occupancy"))
+                if ws.occupancy == 0:  # special case to use round-robin
+                    wp_vals = worker_pool.values()
+                    start = self._n_tasks % n_workers
+                    for i in range(start, n_workers):
+                        if wp_vals[i].occupancy == 0:
+                            ws = wp_vals[i]
```

Comment:
Would select and assign this to the `WorkerState` typed variable. Cython will perform a quick type check and then use this object optimally going forward

```suggestion
                        wp_i = wp_vals[i]
                        if wp_i._occupancy == 0:
                            ws = wp_i
```